radicale/rights.py

```diff
@@ -121,28 +113,31 @@ def authorized(self, user, path, permission):
             {"login": user_escaped, "path": sane_path_escaped})
         if self.rights_type in DEFINED_RIGHTS:
             self.logger.debug("Rights type '%s'", self.rights_type)
-            regex.readfp(StringIO(DEFINED_RIGHTS[self.rights_type]))
+            regex.read_dict(DEFINED_RIGHTS[self.rights_type])
         else:
             self.logger.debug("Reading rights from file '%s'", self.filename)
             if not regex.read(self.filename):
                 self.logger.error(
                     "File '%s' not found for rights", self.filename)
                 return False
 
+        self.logger.debug("Testing user %r for collection %r", user, sane_path)
         for section in regex.sections():
             re_user = regex.get(section, "user")
             re_collection = regex.get(section, "collection")
-            self.logger.debug(
-                "Test if '%s:%s' matches against '%s:%s' from section '%s'",
-                user, sane_path, re_user, re_collection, section)
             # Emulate fullmatch
             user_match = re.match(r"(?:%s)\Z" % re_user, user)
             if user_match:
                 re_collection = re_collection.format(*user_match.groups())
                 # Emulate fullmatch
                 if re.match(r"(?:%s)\Z" % re_collection, sane_path):
-                    self.logger.debug("Section '%s' matches", section)
+                    self.logger.debug("  Section %r: matches %r",
+                                      section, re_collection)
                     return permission in regex.get(section, "permission")
                 else:
-                    self.logger.debug("Section '%s' does not match", section)
+                    self.logger.debug("  Section %r: "
```

Comment:
My change is mimicking the old behaviour:
Old:
```
Reading rights from file '/tmp/radicale/etc/radicale/rights'
Test if 'michael:michael' matches against '^admin$:.*' from section 'admin'
Test if 'michael:michael' matches against '.+$:^michael/(holidays|foobar)$' from section 'public'
Section 'public' does not match
Test if 'michael:michael' matches against '^(.+)-mobil?$:^michael/(calendar|shared-calendar|addr-handy)' from section 'mobile-calendars-rw'
Test if 'michael:michael' matches against '^(.+)-mobil?$:^michael/?$' from section 'mobile-darf-eigene-collections-lesen'
Test if 'michael:michael' matches against '^(.+)-mobil?$:.*' from section 'mobile-darf-sonst-nix'
Test if 'michael:michael' matches against '.+:^michael/.+' from section 'owner-write'
Section 'owner-write' does not match
```
New:
```
Reading rights from file '/tmp/radicale/etc/radicale/rights'
Testing user 'michael' for collection 'michael'
  Section 'admin': User does not match
  Section 'public': Collection does not match '^michael/(holidays|foobar)$'
  Section 'mobile-calendars-rw': User does not match
  Section 'mobile-darf-eigene-collections-lesen': User does not match
  Section 'mobile-darf-sonst-nix': User does not match
  Section 'owner-write': Collection does not match '^michael/.+'
```
You may notice that the old code did report every test, but not whether it matched or not.